File: src/node/s3/createBucket.js
import { S3Client, CreateBucketCommand, waitUntilBucketExists } from "@aws-sdk/client-s3";

const s3Client = new S3Client({
    region: 'us-east-2',
    version: 'latest',
    // profile: 'your-profile',
});

// Search for CreateBucketCommandInput to see input
// to pass to CreateBucketCommand
const Bucket = 'bucket-o-fun'

// Create the Amazon S3 bucket

// data: {Location: "us-east-2"}
// https://docs.aws.amazon.com/zh_cn/AWSJavaScriptSDK/v3/latest/clients/client-s3-control/classes/createbucketcommand.html
const data = await s3Client.send(new CreateBucketCommand({
    Bucket: Bucket
}));

// https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/globals.html#waituntilbucketexists
await waitUntilBucketExists({ s3Client, maxWaitTime: 30, minDelay: 3 }, { Bucket })

File: src/golang/s3/createBucket.go
package main

import (
	"context"
	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/service/s3"
	"log"
	"time"
)

var bucket = "bucket-o-fun"

func main() {
	// Load the Shared AWS Configuration (~/.aws/config)
	cfg, err := config.LoadDefaultConfig(context.TODO() /*, config.WithSharedConfigProfile("my-profile"), config.WithRegion("us-east-2")*/)

	if err != nil {
		log.Fatal(err)
	}

	// Create an Amazon S3 service client
	client := s3.NewFromConfig(cfg)

	result, err := client.CreateBucket(context.TODO(), &s3.CreateBucketInput{
		Bucket: aws.String(bucket),
	})

	if err != nil {
		log.Fatal(err)
	}

	log.Printf("CreateBucket Location: %s", result.Location)

	waiter := s3.NewBucketExistsWaiter(client)
	maxWaitTime := time.Second * 30

	err = waiter.Wait(context.TODO(), &s3.HeadBucketInput{
		Bucket: aws.String(bucket),
	}, maxWaitTime)

	if err != nil {
		log.Fatal(err)
	}
}
